pkg/rke2/psp/psp.go

```diff
@@ -0,0 +1,443 @@
+		if _, err := cs.RbacV1().ClusterRoles().Get(ctx, systemUnrestrictedRoleName, metav1.GetOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				tmpl := fmt.Sprintf(roleTemplate, systemUnrestrictedRoleName, systemUnrestrictedPSPName)
+				if err := deployClusterRoleFromYaml(ctx, cs, tmpl); err != nil {
+					return err
+				}
+			}
+		}
+		if _, err := cs.RbacV1().ClusterRoleBindings().Get(ctx, systemUnrestrictedRoleBindingName, metav1.GetOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				tmpl := fmt.Sprintf(systemUnrestrictedNodesRoleBindingTemplate, systemUnrestrictedRoleBindingName, systemUnrestrictedRoleName)
+				if err := deployClusterRoleBindingFromYaml(ctx, cs, tmpl); err != nil {
+					return err
+				}
+			}
+		}
+		if _, err := cs.RbacV1().ClusterRoleBindings().Get(ctx, systemUnrestrictedSvcAcctRoleBindingName, metav1.GetOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				tmpl := fmt.Sprintf(systemUnrestrictedServiceAcctRoleBindingTemplate, systemUnrestrictedSvcAcctRoleBindingName, systemUnrestrictedRoleName)
+				if err := deployRoleBindingFromYaml(ctx, cs, tmpl); err != nil {
+					return err
+				}
+			}
+		}
+		ns.SetAnnotations(map[string]string{namespaceAnnotationSystemUnrestricted: cisAnnotationValue})
+	}
+	return nil
+}
+
+// kubeConfigExists checks if the kubeconfig file
+// has been written.
+func kubeConfigExists(kc string) bool {
+	info, err := os.Stat(kc)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+// SetPSPs sets the default PSP's based on the mode that RKE2 is running in. There is either CIS or non
+// CIS mode. For CIS mode, the globalRestricted and systemUnrestricted polices and associated roles and
+// bindings are created. For non CIS mode, globalRestricted, system-unrestricted, and global-unrestricted
+// policies and associated roles and bindings are loaded, if applicable.
+//
+// Each PSP has an annotation associated with it that we check to see exists before performing any
+// write operations. The annotation is created upon successful setting of PSPs.
+//
+// Load logic
+//
+// CIS:
+// - If the globalRestricted annotation does not exist, create PSP, role, binding.
+// - If the systemUnrestricted annotation does not exist, create the PSP, role, binding.
+// - If the globalUnrestricted annotation does not exist, check if PSP exists, and if so,
+//   delete it, the role, and the bindings.
+//
+// non-CIS:
+// - If the globalUnrestricted annotation does not exist, then create PSP, role, binding.
+// - If the systemUnrestricted annotation does not exist, then create PSP, role, binding.
+// - If the globalRestricted annotation does not exist, then check if the PSP exists and
+//   if it doesn't, create it. Check if the associated role and bindings exist and
+//   if they do, delete them.
+func SetPSPs(sCtx context.Context, ctx *cli.Context, k8sWrapTransport transport.WrapperFunc) error {
+	const (
+		kubeConfigPath = "/etc/rancher/rke2/rke2.yaml"
```

Comment:
DISCLAIMER: I don't have a solution on-hand. Just noting my concerns at this point. Hopefully we can brainstorm a solution today.

I know we dont have a better technique on hand, but relying on a hard-coded location of the config doesn't really feel right. k3s's server.run() method (and the structs available therein) have much better approaches to similar. For example, `config.ControlConfig.Runtime.KubeConfigAdmin` gives you the config.

Reply:
I agree. I couldn't find any better way at that point.

Reply:
must-have BUT I dont have a better solution. I just want to go over this one with @ibuildthecloud, but he said not until tomorrow

Reply:
We have the `--write-kubeconfig` flag that I could possibly reference but it looks like rke2 explicitly writes that config to that location and this flag will create a symlink to where the user may want to reference it.